### ISSUE TITLE
Cannot install behind firewall

### ISSUE THREAD

**Issue by nightbeacons** [platformsh/platformsh-cli#952]
The CLI installer fails when attempting to install behind a corporate firewall. The installer does not recognize HTTP_PROXY and HTTPS_PROXY env vars.

The issue seems to be with file_get_contents() -- which needs some special handling WRT proxies (See https://stackoverflow.com/questions/31634604/file-get-contents-with-https-requests-via-proxy )

The manual install method detailed at https://github.com/platformsh/platformsh-cli#installing-manually does work (but this is not in the docs for installing the CLI)

Can the installer be adjusted to recognize and use a proxy?

**Comment by pjcdawkins** [comment]
Yes the installer should be adapted to use a proxy.

Do other CLI commands work behind the proxy? Is it definitely `HTTP_PROXY` and `HTTPS_PROXY` uppercase like that? I thought they were lowercase, and the other proxy handling code assumes lowercase:

https://github.com/platformsh/platformsh-cli/blob/d9c4ecca23b16e9f7b7e1c7c5a2d01b13bd4f105/src/Service/Api.php#L449-L469

**Comment by nightbeacons** [comment]
I've just installed the CLI on a CentOS 7 box -- so far, I am running into
proxy problems with other commands, as well.
(And yes, usually I see the lowercase versions of those.)

**Comment by nightbeacons** [comment]
I have verified that
*platform  ssh-key:list*
*does *work with my http_proxy env var setting.  If I find some that
misbehave, I will add them to the ticket.

Thank you!

**Comment by pjcdawkins** [comment]
Could you try the installer from that PR?

```sh
curl -sS https://raw.githubusercontent.com/platformsh/platformsh-cli/d6aea7b82daca139e4ee6414e29ae9f63cfe61a4/dist/installer.php | php
```

**Comment by nightbeacons** [comment]
Hi --
On your Slack channel, 'Damz' suggested the following, which worked.
I am now working with our corporate IT team to allow SSH between your
servers and mine.

Try following the manual installation steps:
https://github.com/platformsh/platformsh-cli#installing-manually

Thank you!

**Comment by pjcdawkins** [comment]
Hi @nightbeacons 

Thanks yes. I mean this version of the installer should work:

https://raw.githubusercontent.com/platformsh/platformsh-cli/d6aea7b82daca139e4ee6414e29ae9f63cfe61a4/dist/installer.php

I wonder if you could test it behind your proxy - with the command above (or download that file and run it with PHP).

The source code changes are here https://github.com/platformsh/platformsh-cli/pull/953/files

**Comment by nightbeacons** [comment]
Sorry -- misunderstood

I d/led the installer and ran from php:

 php installer.php
Platform.sh CLI installer

Environment check
  [*] The "json" PHP extension is installed.
  [*] The "phar" PHP extension is installed.
  [*] Git is installed.
  [*] The "openssl" PHP extension is installed.
  [*] The "pcre" PHP extension is installed.
  [*] The "curl" PHP extension is installed.
  [*] The "pcntl" and "posix" extensions are installed.
  [*] The "allow_url_fopen" setting is on.
  [*] The "apc.enable_cli" setting is off.

Download
  Finding the latest version...
Warning: file_get_contents(https://platform.sh/cli/manifest.json): failed
to open stream: Unable to find the socket transport "http" - did you forget
to enable it when you configured PHP? in /tmp/installer.php on line 309

Error: Failed to download manifest file:
https://platform.sh/cli/manifest.json

**Comment by pjcdawkins** [comment]
Thanks @nightbeacons . I've tried a new fix:

https://raw.githubusercontent.com/platformsh/platformsh-cli/7a4f890f4ea35474b55c27aee34ed749ce3a39c5/dist/installer.php

**Comment by pjcdawkins** [comment]
As @rokkie points out this is probably now fixed by #1037
